src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -222,7 +222,7 @@ public static CommandLineBuilder CancelOnProcessTermination(this CommandLineBuil
                     string debuggableProcessNames = GetEnvironmentVariable(environmentVariableName);
                     if (string.IsNullOrWhiteSpace(debuggableProcessNames))
                     {
-                        context.Console.Error.WriteLine(Resources.Instance.DebugDirectiveExecutableNotSpecified(environmentVariableName, process.ProcessName));
+                        context.Console.Error.WriteLine((builder.Resources ?? Resources.Instance).DebugDirectiveExecutableNotSpecified(environmentVariableName, process.ProcessName));
```

Comment:
There is a subtle change here, that now causes this lambda to capture the builder parameter. Though I don't think it is likely to cause an issue here, it might be cleaner to retrieve the resource by going through the already passed in `context` (as you noted in the description: `context.ParseResult.CommandResult.Resources`). 

The same would apply to the two other usages further down in this method.

I do also agree that the Resources should probably be exposed on `InvocationContext` directly rather than burring them down so deep.

Reply:
Resolved (7da17af1fbd8e26c12d652ac100ad383c4391e34).